cypress/integration/learn/responsive-web-design/claim-cert-from-learn.js

```diff
@@ -0,0 +1,99 @@
+    },
+    {
+      slug: 'build-a-product-landing-page',
+      solution: 'https://codepen.io/moT01/full/qKyKYL/'
+    },
+    {
+      slug: 'build-a-technical-documentation-page',
+      solution: 'https://codepen.io/moT01/full/JBvzNL/'
+    },
+    {
+      slug: 'build-a-personal-portfolio-webpage',
+      solution: 'https://codepen.io/moT01/pen/vgOaoJ'
+    }
+  ]
+};
+
+describe('Responsive Web Design Superblock', () => {
+  before(() => {
+    cy.exec('npm run seed');
+    cy.login();
+    cy.visit('/learn/responsive-web-design');
+  });
+  describe('Before submitting projects', () => {
+    it('should have a card with href "claim-cert-block"', () => {
+      cy.get('a[href="#claim-cert-block"]').scrollIntoView();
+      cy.get('a[href="#claim-cert-block"]').should('be.visible');
+    });
+
+    it('should have an anchor element with the text "Claim Certification", and class "disabled"', () => {
+      cy.get('a.disabled').should('be.visible');
+      cy.get('a.disabled').should('have.text', 'Claim Certification');
+    });
+
+    it('should have an unordered list with class "map-challenges-ul" containing 5 items', () => {
+      cy.get('a[href="#claim-cert-block"]')
+        .parent()
+        .parent()
+        .find('ul.map-challenges-ul')
+        .should('be.visible');
+      cy.get('a[href="#claim-cert-block"]')
+        .parent()
+        .parent()
+        .find('ul.map-challenges-ul')
+        .children()
+        .should('have.length', 5);
+    });
+  });
+  describe('After submitting all 5 projects', () => {
+    before(() => {
+      cy.toggleAll();
+      const { superBlock, block, challenges } = projects;
+      challenges.forEach(({ slug, solution }) => {
+        const url = `/learn/${superBlock}/${block}/${slug}`;
+        cy.visit(url);
+        cy.get('#dynamic-front-end-form')
+          .get('#solution')
+          .type(solution, { force: true, delay: 0 });
+        cy.contains("I've completed this challenge")
+          .should('not.be.disabled')
+          .click();
+        cy.contains('Submit and go to next challenge').click().wait(1000);
+      });
+    });
+    it('should redirect to the Superblock page', () => {
+      cy.url().should('include', `/learn/${projects.superBlock}`);
+    });
+    it('should open the donation modal on the Superblock page', () => {
+      cy.contains('Ask me later').should('be.visible').click().wait(300);
+    });
+    it('should have the hash "claim-cert-block" in the url', () => {
+      cy.url().should('include', '#claim-cert-block');
+    });
+    it('should be able to claim certification', () => {
+      cy.get('[href="/certification/developmentuser/responsive-web-design"]')
+        .should('not.be.disabled')
+        .click();
+    });
+    it('should be able to view the certification', () => {
+      cy.contains('Show Certification').click();
+      cy.url().should(
+        'contain',
+        'certification/developmentuser/responsive-web-design'
+      );
+    });
```

Comment:
```suggestion
        cy.contains('Submit and go to next challenge').click();
        cy.location().should(loc => {
          expect(loc.pathname).to.not.eq(url);
        });
      });
    });
    it('should be possible to claim and view certifications from the superBlock page', () => {
      cy.location().should(loc => {
        expect(loc.pathname).to.eq(`/learn/${projects.superBlock}`);
      });
      cy.get('.donation-modal').should('be.visible');
      cy.contains('Ask me later').click();
      cy.get('.donation-modal').should('not.be.visible');
      // directed to claim-cert-block section
      cy.url().should('include', '#claim-cert-block');
      cy.contains('Claim Certification').should('not.be.disabled').click();
      cy.contains('Show Certification').click();
      cy.location().should(loc => {
        expect(loc.pathname).to.eq(
          '/certification/developmentuser/responsive-web-design'
        );
      });
```
The Cypress best practices recommend [writing larger tests](https://docs.cypress.io/guides/references/best-practices#Creating-tiny-tests-with-a-single-assertion).  Also, explicit waits _generally_ aren't necessary. If you instead use `cy.contains('the thing you are waiting for')` it will wait long enough to see if what you're expecting to happen has happened.

Reply:
Ah right. Sorry - when you mentioned longer tests, I thought it just involved the `describes`.